v2/pkg/engine/datasource/grpc_datasource/execution_plan_field_resolvers_test.go

```diff
@@ -0,0 +1,894 @@
+											},
+										},
+									},
+								},
+								{
+									Name:     "field_args",
+									TypeName: string(DataTypeMessage),
+									JSONPath: "",
+									Message: &RPCMessage{
+										Name: "ResolveCategoryProductCountArgs",
+										Fields: []RPCField{
+											{
+												Name:     "filters",
+												TypeName: string(DataTypeMessage),
+												JSONPath: "p1",
+												Optional: true,
+												Message: &RPCMessage{
+													Name: "ProductCountFilter",
+													Fields: []RPCField{
+														{
+															Name:     "min_price",
+															TypeName: string(DataTypeDouble),
+															JSONPath: "minPrice",
+															Optional: true,
+														},
+														{
+															Name:     "max_price",
+															TypeName: string(DataTypeDouble),
+															JSONPath: "maxPrice",
+															Optional: true,
+														},
+														{
+															Name:     "in_stock",
+															TypeName: string(DataTypeBool),
+															JSONPath: "inStock",
+															Optional: true,
+														},
+														{
+															Name:     "search_term",
+															TypeName: string(DataTypeString),
+															JSONPath: "searchTerm",
+															Optional: true,
+														},
+													},
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+						Response: RPCMessage{
+							Name: "ResolveCategoryProductCountResponse",
+							Fields: []RPCField{
+								{
+									Name:     "result",
+									TypeName: string(DataTypeMessage),
+									JSONPath: "result",
+									Repeated: true,
+									Message: &RPCMessage{
+										Name: "ResolveCategoryProductCountResult",
+										Fields: []RPCField{
+											{
+												Name:     "product_count",
+												TypeName: string(DataTypeInt32),
+												JSONPath: "productCount1",
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+					{
+						DependentCalls: []int{0},
+						ServiceName:    "Products",
+						MethodName:     "ResolveCategoryProductCount",
+						Kind:           CallKindResolve,
+						ResponsePath:   buildPath("categories.productCount2"),
+						Request: RPCMessage{
+							Name: "ResolveCategoryProductCountRequest",
+							Fields: []RPCField{
+								{
+									Name:     "context",
+									TypeName: string(DataTypeMessage),
+									JSONPath: "",
+									Repeated: true,
+									Message: &RPCMessage{
+										Name: "ResolveCategoryProductCountContext",
+										Fields: []RPCField{
+											{
+												Name:        "id",
+												TypeName:    string(DataTypeString),
+												JSONPath:    "id",
+												ResolvePath: buildPath("categories.id"),
+											},
+											{
+												Name:        "name",
+												TypeName:    string(DataTypeString),
+												JSONPath:    "name",
```

Comment:
I think it would be better (and more efficient) to align the formats here with what astjson needs so that we can use astjson at execution without transforming again